### ISSUE TITLE
storage mount failed silently when fusermount not installed

### ISSUE THREAD

**Issue by lhqing** [skypilot-org/skypilot#1927]
Hi SkyPilot team

Report another issue with custom image

**Problem**
On GCP, using custom image to launch VM.
File mount fail silently when then custom image do not have `fusermount` installation.

**Solution**
Have to install fuse in the image `sudo apt install -y fuse`.

`sky launch` output:
```
# hanqingliu @ belmontezf in ~/src/whole_mouse_brain/gcp/sky on git:main x [10:48:41] 
$ sky launch -c jupyter -y sky-jupyter.yaml 
Task from YAML spec: sky-jupyter.yaml
W 05-02 10:51:05 resources.py:860] image_id in resources is experimental. It only supports AWS/GCP.
I 05-02 10:51:09 optimizer.py:617] == Optimizer ==
I 05-02 10:51:09 optimizer.py:640] Estimated cost: $1.4 / hour
I 05-02 10:51:09 optimizer.py:640] 
I 05-02 10:51:09 optimizer.py:712] Considered resources (1 node):
I 05-02 10:51:09 optimizer.py:760] ----------------------------------------------------------------------------------------------
I 05-02 10:51:09 optimizer.py:760]  CLOUD   INSTANCE          vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN   
I 05-02 10:51:09 optimizer.py:760] ----------------------------------------------------------------------------------------------
I 05-02 10:51:09 optimizer.py:760]  GCP     n2d-standard-32   32      128       -              us-west1      1.35          ✔     
I 05-02 10:51:09 optimizer.py:760] ----------------------------------------------------------------------------------------------
I 05-02 10:51:09 optimizer.py:760] 
Running task on cluster jupyter...
I 05-02 10:51:09 cloud_vm_ray_backend.py:3327] Creating a new cluster: "jupyter" [1x GCP(n2d-standard-32, image_id={'us-west1': 'projects/prod-635e/global/images/hanqing-base-debian10'})].
I 05-02 10:51:09 cloud_vm_ray_backend.py:3327] Tip: to reuse an existing cluster, specify --cluster (-c). Run `sky status` to see existing clusters.
I 05-02 10:51:10 cloud_vm_ray_backend.py:1156] To view detailed progress: tail -n100 -f /Users/hanqingliu/sky_logs/sky-2023-05-02-10-51-07-968092/provision.log
I 05-02 10:51:11 authentication.py:199] OS Login is enabled for GCP project prod-635e. Running additional authentication steps.
I 05-02 10:51:15 cloud_vm_ray_backend.py:1480] Launching on GCP us-west1 (us-west1-a)
I 05-02 10:52:00 log_utils.py:89] Head node is up.
I 05-02 10:52:46 cloud_vm_ray_backend.py:1293] Successfully provisioned or found existing VM.
I 05-02 10:52:54 cloud_vm_ray_backend.py:3518] Processing 5 storage mounts.
I 05-02 10:52:54 backend_utils.py:1199] Mounting (to 1 node): gs://ecker-cemba-rs2/ -> /rs2
I 05-02 10:52:58 backend_utils.py:1199] Mounting (to 1 node): gs://ecker-rachel-analysis/ -> /data
I 05-02 10:52:59 backend_utils.py:1199] Mounting (to 1 node): gs://ecker-hanqing-wmb-us-west1/ -> /cemba
I 05-02 10:53:00 backend_utils.py:1199] Mounting (to 1 node): gs://ecker-hanqing-reference/ -> /ref
I 05-02 10:53:01 backend_utils.py:1199] Mounting (to 1 node): gs://ecker-hanqing-analysis/ -> /home/hanliu
I 05-02 10:53:02 cloud_vm_ray_backend.py:2530] Running setup on 1 node.
Warning: Permanently added '34.82.158.36' (ED25519) to the list of known hosts.
...
```

**storage_mounts.log**
```
# hanqingliu @ belmontezf in ~/sky_logs/sky-2023-05-02-10-51-07-968092 [10:53:01]
$ tail -F storage_mounts.log
Mounting done.
bash: warning: here-document at line 37 delimited by end-of-file (wanted `EOF')
gcsfuse already installed. Proceeding...
Mount path /home/hanliu does not exist. Creating...
Mounting  to /home/hanliu with gcsfuse...
2023/05/02 17:53:02.663928 Start gcsfuse/0.41.10 (Go version go1.18.4) for app "" using mount point: /home/hanliu
2023/05/02 17:53:02.673994 Opening GCS connection...
2023/05/02 17:53:02.816030 Mounting file system "ecker-hanqing-analysis"...
daemonize.Run: readFromProcess: sub-process: mountWithArgs: mountWithConn: Mount: mount: exec: "fusermount": executable file not found in $PATH
Mounting done.
```

**Comment by lhqing** [comment]
Related to: https://github.com/skypilot-org/skypilot/issues/1604#issuecomment-1530052350

Seems to me the current custom image feature missing some basic installation check. An expected behavior might be SkyPilot will validate all necessary tools (`rsync`, `fuse` etc) being installed or install them automatically when missing

**Comment by romilbhardwaj** [comment]
Will be closed with #3400.
